ext/standard/string.c

```diff
@@ -2400,12 +2410,22 @@ PHP_FUNCTION(substr_replace)
 			zend_string *orig_str = zval_get_tmp_string(tmp_str, &tmp_orig_str);
 
 			if (from_ht) {
-				while (from_idx < from_ht->nNumUsed) {
-					tmp_from = &from_ht->arData[from_idx].val;
-					if (Z_TYPE_P(tmp_from) != IS_UNDEF) {
-						break;
+					while (from_idx < from_ht->nNumUsed) {
+						tmp_from = &from_ht->arPacked[from_idx];
+						if (Z_TYPE_P(tmp_from) != IS_UNDEF) {
+							break;
+						}
+						from_idx++;
+					}
+				if (HT_IS_PACKED(from_ht)) {
```

Comment:
The HT_IS_PACKED check should go above the while loop you added? There's nothing between the if on line 2420 and the else on line 2421 right now.

```suggestion
				if (HT_IS_PACKED(from_ht)) {
					while (from_idx < from_ht->nNumUsed) {
						tmp_from = &from_ht->arPacked[from_idx];
						if (Z_TYPE_P(tmp_from) != IS_UNDEF) {
							break;
						}
						from_idx++;
					}
```

Reply:
Right. Thanks.